### ISSUE TITLE
template module generating file with excessive content

### ISSUE THREAD

**Issue by PikachuEXE** [ansible/ansible#49012]
##### SUMMARY

Using 2.7.x (2.7.2 latest) the nginx config file generated by template module is different from what's generated before.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

template

##### ANSIBLE VERSION

```paste below
ansible 2.7.2
  config file = /users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg
  configured module search path = ['/users/pikachuexe/projects/spacious/spacious-rails/ansible/library']
  ansible python module location = /usr/local/Cellar/ansible/2.7.2/libexec/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.1 (default, Nov  6 2018, 18:45:35) [Clang 10.0.0 (clang-1000.11.45.5)]
```

##### CONFIGURATION

```paste below
DEFAULT_FILTER_PLUGIN_PATH(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = ['/users/pikachuexe/projects/spacious/spacious-rails/ansible/filter_plugins']
DEFAULT_HOST_LIST(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = ['/users/pikachuexe/projects/spacious/spacious-rails/ansible/inventories/localhost']
DEFAULT_JINJA2_NATIVE(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = True
DEFAULT_MODULE_PATH(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = ['/users/pikachuexe/projects/spacious/spacious-rails/ansible/library']
DEFAULT_ROLES_PATH(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = ['/users/pikachuexe/projects/spacious/spacious-rails/ansible/roles']
DEFAULT_TIMEOUT(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = 30
DEFAULT_UNDEFINED_VAR_BEHAVIOR(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = True
DISPLAY_SKIPPED_HOSTS(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = False
RETRY_FILES_ENABLED(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Source OS: MacOS 10.13.6
Target OS: Ubuntu 18.04.1 LTS (Linux 4.15.0-1023-gcp x86_64)

##### STEPS TO REPRODUCE

- Run the only playbook at https://github.com/PikachuEXE/ansible_27x_template_module_issue_example
- Inspect the files generated by ansible 2.6.x & 2.7.x

##### EXPECTED RESULTS

The file content should be the same as what's generated in 2.6.x

##### ACTUAL RESULTS

Got some extra lines which should not be output

**Comment by ansibot** [comment]
Hi @PikachuEXE, thank you for submitting this issue!

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/files/template.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/template.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mkrizek** [comment]
What version of Jinja2 do you have installed?
What's in
`DEFAULT_FILTER_PLUGIN_PATH(/users/pikachuexe/projects/spacious/spacious-rails/ansible.cfg) = ['/users/pikachuexe/projects/spacious/spacious-rails/ansible/filter_plugins']` ?
Does the issue persist when `jinja2_native` is set to `False`?

I can reproduce the issue with `2.7.0`:
```
--- v26-py3.conf        2018-11-22 08:05:02.040893328 -0500
+++ v27-py3.conf        2018-11-22 08:10:59.285071728 -0500
@@ -6958,4 +6958,4 @@

     return        444;
   }
-}
+}
\ No newline at end of file
```

but not with `2.7.1` and `2.7.2`.

Can you post your diff (like above) here?

Thanks!

needs_info

**Comment by PikachuEXE** [comment]
`/users/pikachuexe/projects/spacious/spacious-rails/ansible/filter_plugins` folder used to store old version of `combine` filter (before it was introduced) and is now empty. Please ignore it

jinja2 version:
```
pip show Jinja2
Name: Jinja2
Version: 2.9.6
Summary: A small but fast and easy to use stand-alone template engine written in pure python.
Home-page: http://jinja.pocoo.org/
Author: Armin Ronacher
Author-email: armin.ronacher@active-4.com
License: BSD
Location: /usr/local/lib/python2.7/site-packages
Requires: MarkupSafe
Required-by: Sphinx, gixy, Flask
```

Issue is gone after setting `jinja2_native` to `False`
But it's required to be `True` to solve #38148

**Comment by PikachuEXE** [comment]
Sorry forgot to attach the diff

[nginx.conf.diff.txt](https://github.com/ansible/ansible/files/2609581/nginx.conf.diff.txt)

**Comment by mkrizek** [comment]
> pip show Jinja2
Name: Jinja2
Version: 2.9.6

OK, so this explains it, `jinja2_native` requires `Jinja2 >= 2.10`, please see https://docs.ansible.com/ansible/2.7/reference_appendices/config.html#default-jinja2-native

As per above, I am going to close this.

If you have further questions please stop by IRC or the mailing list:

IRC: #ansible on irc.freenode.net
mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by mkrizek** [comment]
Actually, this should fall back to `jinja2_native = False`.

**Comment by mkrizek** [comment]
> Actually, this should fall back to jinja2_native = False.

Which it does.

OK, so this is duplicate of https://github.com/ansible/ansible/issues/46113.

After looking at your `pip show Jinja2` output more closely, it looks like you ran that command on the managed node (Ubuntu in your case) while templating is done on the controller (MacOS). So it seems like you do have Jinja2 `2.10` installed on your controller. Unfortunately, there is a bug there that is fixed upstream and so you need to run unreleased version of Jinja2 (https://github.com/pallets/jinja - master branch) on your controller.

**Comment by mkrizek** [comment]
Fixed upstream https://github.com/pallets/jinja/pull/799. Closing.
